books/RayTracingTheRestOfYourLife.html

```diff
@@ -47,22 +47,51 @@
 
 
 
-
 A Simple Monte Carlo Program
 ====================================================================================================
+<div class='together'>
+Let’s start with one of the simplest Monte Carlo programs. If you're not familiar with Monte Carlo
+programs, then it'll be good to pause and catch you up. There are two kinds of randomized
+algorithms: Monte Carlo algorithms and Las Vegas algorithms. Randomized algorithms can be found
+everywhere in computer graphics, so getting a decent foundation isn't a bad idea. A randomized
+algorithm is an algorithm that uses some amount of randomness in it's computation. A Las Vegas (LV)
+random algorithm always produces the correct result, whereas a Monte Carlo (MC) algorithm _may_
+produce a correct result (and frequently gets it wrong!). But for especially complicated problems
+such as ray tracing, we may not place as huge a priority on being perfectly correct as on getting an
+answer in a reasonable amount of time. LV algorithms will eventually arrive at the correct result,
+but we can't make too many guarantees on how long it will take to get there. The classic example of
+an LV algorithm is the $quicksort$ sorting algorithm. The quicksort algorithm will always complete
+with a fully sorted list, but, the time it takes to complete is random. Another good example of an
+LV algorithm is the code that we use to pick a random point in a unit sphere:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    inline vec3 random_in_unit_sphere() {
+        while (true) {
+            auto p = vec3::random(-1,1);
+            if (p.length_squared() >= 1) continue;
+            return p;
+        }
+    }
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [las-vegas-algo]: <kbd>[vec3.h]</kbd> A Las Vegas algorithm]
 
-Let’s start with one of the simplest Monte Carlo (MC) programs. MC programs give a statistical
-estimate of an answer, and this estimate gets more and more accurate the longer you run it. This
-basic characteristic of simple programs producing noisy but ever-better answers is what MC is all
-about, and it is especially good for applications like graphics where great accuracy is not needed.
+This code will always eventually arrive at a random point in the unit sphere, but we can't say
+beforehand how long it'll take. It may take only 1 `while` loop, it may take 2, 3, 4, or even
+longer. Whereas, MC programs give a statistical estimate of an answer, and this estimate gets more
+and more accurate the longer you run it. Which means that at a certain point, we can just decide
+that the answer is accurate _enough_ and call it quits. This basic characteristic of simple programs
+producing noisy but ever-better answers is what MC is all about, and is especially good for
```

Comment:
With regard to "ever better", would it be helpful to say that with each iteration, MC produces answers that are the same or better, but never worse?

Reply:
This is not true.
If I had a coin that lands heads 60% of the time, and tails 40% of the time.

And I didn't know what the odds were, but I wanted to figure it out by flipping the coin a lot,
If I flipped the coin 10 times and got:

HTHHTHTHHT

Then, so, far, I have 6 heads and 4 tails.
My MC so far would lead me to believe that it's 60-40.

But let's say that I continue, and get:

HTHHTHTHHT HHH

Then I would have 9 heads and 4 tails,
My MC would then lead me to believe that it's 69-31
(Which is less accurate)

MC tends toward exactness, but constant improvement is not guaranteed.

For a high dimensional problem like Raytracing, you can practically guarantee that more samples will always be a more accurate (and precise) answer, but this isn't true in general.

Reply:
Also, the statement:
"
This basic characteristic of simple programs
producing noisy but ever-better answers is what MC is all about, and is especially good for
applications like graphics where great accuracy is not needed.
"

Was Shirley's from the original text and it felt weird rewriting an entire section. I like this portion and figured it worked well.

Reply:
And MC algorithms can happen over continuous testing domains, where the idea of a discrete "iteration" doesn't really hold.
So, the sentence as-is, I found to be rather succinct and useful.